spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -1351,6 +1378,15 @@ protected String getRemoteFilename(String remoteFilePath) {
 		}
 	}
 
+	private FileExistsMode resolveFileExistsMode(Message<?> message) {
+		if (this.fileExistsModeExpression != null) {
+			EvaluationContext evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+			evaluationContext.setVariable("fileExistsMode", this.fileExistsMode);
+			return this.fileExistsModeExpression.getValue(evaluationContext, message, FileExistsMode.class);
```

Comment:
I also think we should be able to allow to evaluate expression to just string and then try to resolve enum manually.
This way our contract for this expression would be much friendly for end-users.

Reply:
I should have considered user-friendliness more! The expression can now return either a `FileExistsMode` enum value or a String representation (case-insensitive).  Would it be better not to support case-insensitive matching?